src/recipes/tooling-custom-resources.md

```diff
@@ -0,0 +1,40 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Custom Resources
+
+Custom `Resource`s are exposed for the end user to use within their development, but cannot provide any functionality automatically.
+For example, you can create a custom `AudioStream` type that handles a new and interesting audio file type.
+
+
+## Registering a Resource
+
+This workflow is incredibly similar to the [Hello World Example](/intro/hello-world.md)
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, base=Resource)] // use tool!
+struct ResourceType {
+    #[base]
+    base: Base<Resource>,
+}
+
+#[godot_api]
+impl IResource for ResourceType {}
+
+#[godot_api]
+impl ResourceType {
+    #[func]
+    fn foo(&mut self) {
+        godot_print!("bar");
+    }
+}
```

Comment:
Can you think of any methods that are typical for resources?

Otherwise it's probably better to omit this `impl` block. Syntax to declare functions has been covered before, so this doesn't add any new information.